writing-your-first-es2015-javascript-module.md

```diff
@@ -0,0 +1,140 @@
+<!--
+headline: Writing Your First ES2015 JavaScript Module
+description: dead simple guide to writing your first ES2015 module using Babel.
+author: Ben Coe
+datePublished: 2015-11-29
+twitter: benjamincoe
+github: bcoe
+-->
+
+> ES2015 introduces a many slick language features (arrow functions, classes, template strings), but it
+  can be overwhelming figure out where to begin with it!
+
+The library [nyc](https://github.com/bcoe/nyc) grew out of my frustration with how complicated it was to instrument JavaScript projects with test coverage:
+
+```js
+mocha --ui exports --require blanket -R mocoverage
+```
+
+incantations like this were the norm in my _package.json_'s script stanza.
+
+I dreamt of a simpler world, one where you could write `nyc mocha` and have test coverage work like magic.
+
+_this world turned out to be awesome!_
+
+Recently an [issue](https://github.com/bcoe/nyc/issues/53) was opened on nyc requesting support for Babel.
+This brought my attention to a problem similar to the one that nyc was originally built to solve ...
+
+There are many different approaches folks use to wire up ES2015 to ES5 transpilation, the methods use a variety
+of different build tools (_gulp_, _grunt_, _webpack_, _browserify_), and the paradox of choice can be overwhelming.
+
+In the process of adding Babel support to nyc I took notes, my goal being to outline a
+_dead simple_ approach for writing ES2015 modules that work on older versions of Node.js: using
+npm scripts instead of build tools, keeping library dependencies to a minimum, and avoiding
+build steps where possible.
```

Comment:
Perhaps:

In the process of adding Babel support to nyc I took notes, my goal being to outline a
_dead simple_ approach for writing ES2015 modules that work on older versions of Node.js. An approach that emphasizes:

* using npm scripts rather than of build tools.
* keeping library dependencies to a minimum.
* and, where possible, avoiding unnecessary build steps.